code/game/objects/machinery/floodlight.dm

```diff
@@ -136,6 +136,206 @@
 /obj/machinery/floodlight/landing/attackby()
 	return
 
+/obj/machinery/floodlightcombat
+	name = "Armoured floodlight"
+	icon = 'icons/obj/machines/floodlight.dmi'
+	icon_state = "floodlightcombat_off"
+	anchored = TRUE
+	density = TRUE
+	/// Determines how much light does the floodlight make , every light tube adds 4 tiles distance.
+	var/brightness = 0
+	/// Turned ON or OFF? Used for the overlays and power usage.
+	var/on = 0
+	/// Used to show if the object is tipped
+	var/tipped
+	resistance_flags = UNACIDABLE
+	idle_power_usage = 50
+	active_power_usage = 2500
+	wrenchable = TRUE
+
+/obj/machinery/floodlightcombat/wrench_act(mob/living/user, obj/item/I)
+	. = ..()
+	to_chat(user , "<span class='notice'>You begin wrenching [src]'s bolts")
+	playsound( loc, 'sound/items/ratchet.ogg', 60, FALSE)
+	if(!do_after(user, 2 SECONDS, TRUE, src))
+		return FALSE
+	if(anchored)
+		to_chat(user , "<span class='notice'>You unwrench [src]'s bolts")
+		anchored = 0
+	else
+		to_chat(user , "<span class='notice'>You wrench down [src]'s bolts")
+		anchored = 1
+	set_light(0)
```

Comment:
best to set all the arguments

`/atom/proc/set_light(l_range, l_power, l_color = NONSENSICAL_VALUE)`